README.md

```diff
@@ -62,6 +62,15 @@ Please note, that the latter one requires root privileges.
 By default, sxiv is installed using the prefix "/usr/local", so the full path
 of the executable will be "/usr/local/bin/sxiv".
 
+Running make will automatically detect if libexif and libgif are available and
+enable them if so. CLI arguments will override any automatic detection.
+For example:
+
+	$ make HAVE_LIBGIF=0
```

Comment:
Lets say you have already compiled (without the flag), it will not recompile if you pass `HAVE_LIBGIF=0`, instead it will print `make: Nothing to be done for 'all'`. You would need to run `make clean` explicitly.
Idk if the makefile should check for this, but since you are running make with a new flag, I feel like it should recompile.

Reply:
This might be an inherent problem with how make works. It doesn't really track the variables, only the dependencies.

If we can trigger a rebuild on new flags that would be ideal. But I'm not sure if it's possible.

Reply:
I suspected that it wasnt possible, but wanted to point out to see if others have any idea (or even say that its expected behavior).

Reply:
I wouldn't say its _expected_ but rather a limitation of `make`. @TAAPArthur seems to know more about make than I do, maybe he has some trick up his sleeve for this?

Reply:
The only thing I can think of is writing the variables to a file and checking the difference every time `make` is run. But I think that's too hacky and we should just advise running `make clean` first.